framework/src/modules/interoperability/mainchain/commands/recover_message.ts

```diff
@@ -91,9 +91,11 @@ export class RecoverMessageCommand extends BaseInteroperabilityCommand<Mainchain
 			}
 		}
 
-		// Check that the CCMs are still pending. We can check only the first one,
-		// as the idxs are sorted in ascending order.
-		if (idxs[0] < terminatedOutboxAccount.partnerChainInboxSize) {
+		// as the idxs are sorted in ascending order. Note that one must unset the most significant
+		// bit of an encoded index in idxs in order to get the position in the tree.
+		// See https://github.com/LiskHQ/lips/blob/main/proposals/lip-0031.md#proof-serialization.
+		const firstPosition = parseInt(idxs[0].toString(2).slice(1), 2);
```

Comment:
sorry to jump in here, but it's related to the other PR I'm currently reviewing. We should have a unit test for the edge case `idxs = [0]`, i.e. `idxs[0]`does not follow the format of having a prepended `1`. The verify function should fail in this case.

Reply:
Updated